modules/infra/api/iface.c

```diff
@@ -126,6 +126,28 @@ static struct api_out iface_set(const void *request, void ** /*response*/) {
 	return api_out(0, 0);
 }
 
+static void broadcast_iface_event(iface_event_t event, struct iface *iface) {
+	struct gr_infra_iface_get_resp *r;
+	struct gr_api_notification *n;
+
+	n = calloc(1, sizeof(*n) + sizeof(*r));
+	if (n == NULL) {
+		LOG(ERR, "broadcast_iface_event %s", strerror(errno));
+		return;
+	}
+	n->type = (uint32_t)event;
+	n->payload_len = sizeof(*r);
+	r = (struct gr_infra_iface_get_resp *)&n[1];
+
```

Comment:
Instead of malloc here, you could define an inline unnamed structure:

```c
struct {
	struct gr_api_notification n;
    	struct gr_infra_iface_get_resp r;
} data;

data.n.type = (uint32_t)event;
data.n.playload_len = sizeof(data.r);
iface_to_api(&data.r.iface, iface);
gr_api_push_notification(&n);
```